File: java/src/main/java/com/ausregistry/jtoolkit2/se/AeDomainObjectType.java
package com.ausregistry.jtoolkit2.se;

/**
 * Represents the aedomain-1.0 EPP extension object.
 * @deprecated Should instead use {@link ExtendedObjectType#AE_DOMAIN}.
 */
class AeDomainObjectType implements ObjectType {
    private static final long serialVersionUID = -543331887000261081L;

    public String getName() {
        return "aedom";
    }

    public String getURI() {
        return "urn:X-ae:params:xml:ns:aedomain-1.0";
    }

    public String getSchemaLocation() {
        return "urn:X-ae:params:xml:ns:aedomain-1.0 aedomain-1.0.xsd";
    }

    public String getIdentType() {
        return "name";
    }
}

File: cpp/se/AeDomainObjectType.cpp
#include "AeDomainObjectType.hpp"

#include <string>

const std::string& AeDomainObjectType::getName() const {
    static const std::string name = "aedom";
    return name;
}

const std::string& AeDomainObjectType::getURI() const {
    static const std::string uri = "urn:X-ae:params:xml:ns:aedomain-1.0";
    return uri;
}

const std::string& AeDomainObjectType::getSchemaLocation() const {
    static const std::string schemaLocation =
            "urn:X-ae:params:xml:ns:aedomain-1.0 aedomain-1.0.xsd";
    return schemaLocation;
}

const std::string& AeDomainObjectType::getIdentType() const {
    static const std::string ident = "name";
    return ident;
}

